test-config.md

```diff
@@ -0,0 +1,56 @@
+Format of the Test Configuration File
+=====================================
+
+The tests are configured in the file `config.json`.
+
+TODO: Describe the header and test vectors
+
+## Scripts
+
+The configuration file can specify a number of `scripts`. Each script describes a particular scenario of a group's evolution, specified as a sequence of **actions** (e.g. sending a proposal).
+
+### Actors
+Each action is performed by an **actor** (e.g. alice). At each point in time, each actor has at most one MLS client who participates in at most one MLS session. A client and a session for an actor are created in one of two ways:
+
+1. When the actor executes the `createGroup` action, a client with a session is created.
+2. When the actor executes the `createKeyPackage` action, a client without a session is created. When later the actor executes the joinGroup action, a session for the client is created.
+
+Note: Creating a new client / session silently overwrites the first one.
+
+
+### Actions
```

Comment:
You're missing the `verifyStateAuth` action, which is rather important, since it's the one that confirms interop!

In general, you've interpreted the transcript storage parts of the test runner a little differently than I intended.  I added the `storeMessage` bits as an internal state-tracking mechanism, so that you could describe scenarios where the output from prior steps.  Actual interop was to be confirmed with the `verifyStateAuth` action, which confirms that all the actors are in the same state; the development of a transcript was incidental.  You seem to be treating the transcript as the artifact to be used for verifying interop, in a more manual way than `verifyStateAuth`.  We should align on which we're doing here.

Reply:
Isn't the fact that clients can talk at all enough to confirm interop? If client A can receive any packet from client B, then for sure they will derive the same epoch authenticators? The reason is that they must use the same secrets from the key schedule to decrypt the packet or verify the membership tag.

I did understand your intention for the transcript. On the other hand, I found it useful for debugging so I mentioned it. As said above, the main artifact verifying interop in my mind was the fact that client accept each others packets. The transcript _could maybe_ become an artifact if we define how to derandomize implementations, but this seems hard.

Reply:
Checking the epoch authenticator verifies a subset of the functionality that checking send/receive does.  Checking the epoch authenticator checks that you've got TreeKEM and the key schedule right.  Checking send/receive also verifies the secret tree / sender ratchets.  Checking the epoch authenticator is a little lighter-weight, in that it's a single RPC to each client.  Maybe we don't need it.

But at the end of the day, we want the test runner script to know whether interop succeeded or failed for a given script.   For that, `verifyProtectUnprotect` seems better than having `protect` and `unprotect` methods.

Agreed that derandomizing implementations seems hard, and arguably inadvisable, since you wouldn't want that machinery to leak into production.